crates/subspace-archiving/tests/integration/merkle_tree.rs

```diff
@@ -1,12 +1,11 @@
-use std::iter;
 use subspace_archiving::merkle_tree::{MerkleTree, MerkleTreeWitnessError};
-use subspace_core_primitives::{crypto, Piece, Sha256Hash};
+use subspace_core_primitives::{crypto, Piece, Sha256Hash, PIECE_SIZE};
 
 #[test]
 fn merkle_tree() {
     let number_of_pieces = 16_usize;
-    let pieces: Vec<Piece> = iter::repeat_with(rand::random)
-        .take(number_of_pieces)
+    let pieces: Vec<Piece> = (0..number_of_pieces)
+        .map(|_| [rand::random(); PIECE_SIZE].into())
```

Comment:
This is incorrect, this generates an array filled with the same byte value:
```suggestion
    let pieces: Vec<Piece> = iter::repeat_with(|| rand::random::<[u8; PIECE_SIZE]>().into())
        .take(number_of_pieces)
```

Reply:
I must be drunk :D